spec/javascripts/app/views/location_stream_spec.js

```diff
@@ -0,0 +1,39 @@
+describe("app.views.LocationMap", function() {
+  beforeEach(function(){
+    this.post = factory.post();
+    this.view = new app.views.LocationStream({model : this.post});
+  });
+
+  describe("toggleMap", function() {
+    context("with location provided", function() {
+      beforeEach(function(){
+        this.post.set({location : factory.location()}); // set location
+        this.view.render();
+        console.log(this.view.$el.find(".mapContainer")[0]);
+      });
+
+      it("should contain a map container", function() {
+        expect(this.view.$el[0]).toContainElement(".mapContainer");
+      });
+
+      it("should initialize map", function() {
+        expect(this.view.$el.find(".mapContainer")[0]).toHaveClass("empty");
+        this.view.toggleMap();
+        expect(this.view.$el.find(".mapContainer")[0]).not.toHaveClass("empty");
+      });
+      /*
+       * does not work .. not sure why
```

Comment:
The function call initializes the map like in the test above. The `display: block` to `display: none` change only happens after the map has been initialized so you will have to initialize the map first by calling `this.view.toggleMap()` at the beginning of your test and only test the behaviour of the second call.